src/sdk/noobaa_s3_client/noobaa_s3_client.js

```diff
@@ -0,0 +1,148 @@
+/* Copyright (C) 2023 NooBaa */
+'use strict';
+
+const { S3ClientSDKV2 } = require('./noobaa_s3_client_sdkv2');
+const url = require('url');
+const { S3 } = require("@aws-sdk/client-s3");
+const { NodeHttpHandler } = require("@aws-sdk/node-http-handler");
+const dbg = require('../../util/debug_module')(__filename);
+dbg.set_process_name('noobaa_s3_client_by_sdk');
+const config = require('../../../config');
+const http_utils = require('../../util/http_utils');
+
+// The params are the AWS SDK V2 params.
+// params = a map of parameters that are passed to the constructor of S3 Object in order to to bind to every request
+function get_s3_client_v3_params(params) {
+    let s3_client;
+
+    // we use config.AWS_SDK_VERSION_3_DISABLED as workaround in case we want to use only AWS SDK V2
+    if (should_use_sdk_v2(params) || config.AWS_SDK_VERSION_3_DISABLED) {
+        change_s3_client_params_to_v2_structure(params);
+        s3_client = new S3ClientSDKV2(params);
+    } else {
+        s3_client = new S3(params);
+    }
+    return s3_client;
+}
+
+// Here we pack all the functions that would help us decide which version of the sdk should we use
+// How we choose the AWS SDK version for s3 Client?
+// 1. signatureVersion defined? if it is 'v2' => AWS SDK V2
+// 2. maxRedirects defined? if it is a non-negative number => AWS SDK V2
+// 3. AWS SDK V3
+function should_use_sdk_v2(params) {
+    if (params.signatureVersion) {
+        const use_sdk_v2_by_signature_version = should_use_sdk_v2_by_signature_version(params.signatureVersion);
+        if (use_sdk_v2_by_signature_version) return use_sdk_v2_by_signature_version;
+    }
+
+    if (params.maxRedirects) {
+        const use_sdk_v2_by_max_redirects = should_use_sdk_v2_by_max_redirects(params.maxRedirects);
+        return use_sdk_v2_by_max_redirects;
+    }
+
+    return false;
+}
+
+// signature version v2 is only supported in AWS SDK V2
+// v2: The signature version to sign requests with (overriding the API configuration): 'v2' or 'v4'
+// v3: Deprecated, only supports signature 'v4'.
+function should_use_sdk_v2_by_signature_version(signature_version) {
+    return signature_version === 'v2';
+}
+
+// max redirects is only supported in AWS SDK V2
+// v2: The maximum amount of redirects to follow for a service request (Integer, default is 10).
+// v3: Deprecated. SDK does not follow redirects to avoid unintentional cross-region requests.
+function should_use_sdk_v2_by_max_redirects(max_redirects) {
+    return max_redirects >= 0;
+}
+
+// we will use this function in debugging to print the class type easily
+function get_sdk_class_str(s3) {
+    if (is_instance_is_S3ClientSDKV2(s3)) return 'S3 Client SDK V2';
```

Comment:
similar comment here. there is no need to wrap each condition in a function 

```suggestion
    if (s3 instanceof S3ClientSDKV2) return 'S3 Client SDK V2';
```